### ISSUE TITLE
Error referring to union class attribute within classmethod

### ISSUE THREAD

**Issue by Andrej730** [microsoft/pyright#9277]
**Describe the bug**
See example below, `cls.SUPPORTED_TYPES` doesn't seem to be recognized as `Geometry.SUPPORTED_TYPES`.
Current workaround is to explicitly use `Geometry.SUPPORTED_TYPES`.

**Code or Screenshots**

```python
class A:
    x: int
class B:
    x: int
class C: ...

class Object:
    data: A | B | C | None



class Geometry:
    SUPPORTED_TYPES = A | B

    @classmethod
    def test(cls, obj: Object):
        # Cannot access attribute "x" for class "C"
        if ((data:=obj.data) and isinstance(data, cls.SUPPORTED_TYPES) and data.x == 5):
            pass
```

![Image](https://github.com/user-attachments/assets/b28e5101-a716-4962-9931-8d23ba7a1d16)

**VS Code extension or command-line**
VS Code Extension

**Comment by erictraut** [comment]
Pyright is working as intended here. The variable `SUPPORTED_TYPES` is assigned the value `A | B` here. The `|` operator creates an instance of `UnionType`. However, you're using it outside of a type expression, so it is not interpreted as a type by a static type checker. 

If you want type narrowing to work in this case, you should assign `SUPPORTED_TYPES` a value of `(A, B)`. The inferred type of `SUPPORTED_TYPES` will then be `tuple[type[A], type[B]]`, and type narrowing will work correctly.

**Comment by Andrej730** [comment]
But why using `Geometry.SUPPORTED_TYPES` works then, isn't it the same?
